src/validators/currency.validators.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from "zod";
+
+// export const AvailableCurrencyActionsSchema = z.object({
+//   currencyIds: z.array(z.string()),
+// });
+
+export const DefaultCurrencySchema = z.object({
+  currencyId: z.string(),
+});
+
+export const DeleteCurrecncySchema = z.object({
+  currencyIds: z.array(z.string()),
+})
```

Comment:
`import { z } from "zod";

const CurrencySchema = z.object({
  id: z.string(),
  value: z.number(),
});

const CurrencyIdSchema = CurrencySchema.transform((args) => args.id);

export const AvailableCurrencyActionsSchema = z.object({
  currencyIds: CurrencyIdSchema.array(),
});
export const DefaultCurrencySchema = z.object({
  currencyId: CurrencyIdSchema,
});

export const DeleteCurrecncySchema = z.object({
  currencyIds: CurrencyIdSchema.array(),
});

export const AvailabelCurrencySchema = z.object({
  availableCurrencies: CurrencySchema.array(),
});
`



may be infering from a single schema would be a better idea and not repeating 

lets discuss on this tommrow